packet/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -6,15 +6,16 @@ systemd:
       dropins:
         - name: 40-etcd-cluster.conf
           contents: |
+            [Unit]
+            Requires=coreos-metadata.service
+            After=coreos-metadata.service
             [Service]
+            EnvironmentFile=/etc/kubernetes/etcd.config
```

Comment:
This environment file is sourced in the same unit where it is written (via `ExecStartPre`). Doesn't this make the unit to fail because it will try to read the file before it is written?

According to man systemd.exec(5):
> The files listed with this [EnvironmentFile=] directive will be read shortly before the process is executed (more specifically, after all processes from a previous unit state terminated. This means you can generate these files in one unit state, and read it with this option in the next).

Reply:
I am not sure, I understand what you mean here?

The env var file is created in `create-etcd-private-ip-env-file.service` and consumed in `etcd-member.service`. So this works out properly.

Reply:
@alban maybe your comment was before the code change to what you can see now? I'm not sure I follow, if not :)

Reply:
Yes, it is fine in this new version :)

Reply:
Cool. Will mark as solved, then :)